### PR TITLE
tests: Bluetooth: Audio: Cap broadcast reception stop bsim tests

### PR DESCRIPTION
**Pull request by kruithofa**
Add a babblesim test for the CAP broadcast reception stop procedure

### CODE REVIEW

File: tests/bsim/bluetooth/audio/src/cap_acceptor_test.c [hunk 1]
```diff
@@ -401,9 +401,15 @@ static int bis_sync_req_cb(struct bt_conn *conn,
...
+	uint32_t total_bis = 0U;
+
 	broadcaster_broadcast_id = recv_state->broadcast_id;
-	if (bis_sync_req[0] != 0) {
+
+	for (int i = 0; i < recv_state->num_subgroups; i++) {
+		total_bis |= bis_sync_req[i];
+	}
+
+	if (total_bis != 0U) {
```
Comment (Thalley) [inline tests/bsim/bluetooth/audio/src/cap_acceptor_test.c:412]: `total_bis` should be used in `create_and_sync_sink` to determine which BIS to sync to. 

Keep in mind that the value of `bis_sync_req[i]` may be `BT_BAP_BIS_SYNC_NO_PREF`

This is arguably outside the scope of this PR, as the existing test is not using the sync value from the broadcast assistant.

File: tests/bsim/bluetooth/audio/src/cap_acceptor_test.c [hunk 2]
```diff
@@ -1039,10 +1046,12 @@ static void test_cap_acceptor_broadcast_reception(void)
 {
 	static struct bt_bap_stream *bap_streams[ARRAY_SIZE(broadcast_sink_streams)];
 	size_t stream_count;
+	size_t err;
```
Comment (Thalley) [inline tests/bsim/bluetooth/audio/src/cap_acceptor_test.c:1049]:
```suggestion
	int err;
```

File: tests/bsim/bluetooth/audio/src/cap_commander_test.c [hunk 3]
```diff
@@ -580,7 +601,6 @@ static void cap_device_found(const bt_addr_le_t *addr, int8_t rssi, uint8_t type
 		return;
 	}
 
-	printk("Stopping scan\n");
```
Comment (Thalley) [inline tests/bsim/bluetooth/audio/src/cap_commander_test.c:583]: Why was this removed?

File: tests/bsim/bluetooth/audio/src/cap_commander_test.c [hunk 4]
```diff
@@ -607,7 +627,6 @@ static void scan_and_connect(void)
 		return;
 	}
 
-	printk("Scanning successfully started\n");
```
Comment (Thalley) [inline tests/bsim/bluetooth/audio/src/cap_commander_test.c:610]: Why was this removed?

**Review by Thalley — CHANGES_REQUESTED** [review]
A few minor comments, otherwise LGTM

File: tests/bsim/bluetooth/audio/src/cap_acceptor_test.c [hunk 5]
```diff
@@ -401,14 +401,23 @@ static int bis_sync_req_cb(struct bt_conn *conn,
...
+	}
+
+	if (total_bis != 0U) {
 		SET_FLAG(flag_bis_sync_requested);
 	} else {
 		UNSET_FLAG(flag_bis_sync_requested);
 	}
 
+	/* total_bis is the new bis-sync setting to use when calling create_and_sync_sink */
+	bis_index_bitfield = total_bis;
```
Comment (Thalley) [inline tests/bsim/bluetooth/audio/src/cap_acceptor_test.c:419]: This will get overwritten by `base_recv_cb`. 

We should either use the bitfield from here, or from `base_recv_cb`, but using both like this will make it timing dependent on which is used

**Review by Thalley — CHANGES_REQUESTED** [review]

File: tests/bsim/bluetooth/audio/src/cap_acceptor_test.c (hunk 5 above)
Comment (kruithofa) [reply tests/bsim/bluetooth/audio/src/cap_acceptor_test.c:419]: I'll revert, so that we use the bitfield from base_recv_cb

**Review by Thalley — APPROVED** [review]

**Review by larsgk — APPROVED** [review]
